src/sentry/sentry_apps/utils/errors.py

```diff
@@ -7,43 +8,41 @@ class SentryAppErrorType(Enum):
     SENTRY = "sentry"
 
 
-# Represents a user/client error that occured during a Sentry App process
-class SentryAppError(Exception):
-    error_type = SentryAppErrorType.CLIENT
-    status_code = 400
+class ErrorContext(TypedDict, total=False):
+    # Info that gets sent only to the integrator via webhook
+    webhook_context: dict[str, Any]
+    # Info that gets sent to the end user via endpoint Response AND sent to integrator
+    public_context: dict[str, Any]
+
+
+class SentryAppBaseError(Exception):
+    error_type: SentryAppErrorType
+    status_code: int
 
     def __init__(
         self,
-        error: Exception | None = None,
+        message: str,
         status_code: int | None = None,
+        extras: ErrorContext | None = None,
     ) -> None:
-        if status_code:
-            self.status_code = status_code
+        self.status_code = status_code or self.status_code
+        self.extras = extras or {}
+        self.message = message
```

Comment:
I think this was in an earlier PR, but where do we unpack these again into a response again?

Reply:
we're unpacking [here](https://github.com/getsentry/sentry/pull/83331/files#diff-2534681f2d7d9a4677aa2967197ce0400df2ee1a938a0d24745ba429d3558654R134-R135) in the spike.

For the PR to implement the actual design it's [here](https://github.com/getsentry/sentry/pull/83355/files#diff-2534681f2d7d9a4677aa2967197ce0400df2ee1a938a0d24745ba429d3558654R131-R135)